src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs

```diff
@@ -73,10 +84,84 @@ public static string Encode(string arg)
         /// <exception cref="ArgumentOutOfRangeException">offset or length is negative OR offset plus length is greater than the length of inArray.</exception>
         public static string Encode(byte[] inArray, int offset, int length)
         {
-            if (inArray == null)
-                throw LogHelper.LogArgumentNullException("inArray");
+            _ = inArray ?? throw LogHelper.LogArgumentNullException("inArray");
+
+            if (length < 0)
+            {
+                throw LogHelper.LogArgumentException<ArgumentException>(nameof(length), $"{nameof(length)} must be greater or equal than zero");
+            }
+
+            if (offset < 0)
+            {
+                throw LogHelper.LogArgumentException<ArgumentException>(nameof(offset), $"{nameof(offset)} must be greater or equal than zero");
+            }
+
+            if (length == 0)
+            {
+                return string.Empty;
+            }
+
+            if (inArray.Length < offset)
+            {
+                throw LogHelper.LogArgumentException<ArgumentException>(nameof(offset), $"Invalid value for {nameof(offset)} parameter");
```

Comment:
Its good practice to echo the value that was invalid.
So this message could read: "Invalid value for (nameof(offset)) parameter, was: '{offset}'."

Note we tend to put single quotes around such values as it helps identifying the value.